font-types/src/path.rs

```diff
@@ -0,0 +1,24 @@
+/// Interface for accepting a sequence of path commands.
+///
+/// This is a general abstraction to unify ouput for processes that decode and/or
+/// transform outlines.
+///
+/// The type parameter `T` represents the coordinate type.
+pub trait PathSink<T> {
+    /// Emit a command to begin a new subpath at (x, y).
+    fn move_to(&mut self, x: T, y: T);
+
+    /// Emit a line segment from the current point to (x, y).
+    fn line_to(&mut self, x: T, y: T);
+
+    /// Emit a quadratic bezier segment from the current point with a control
+    /// point at (cx0, cy0) and ending at (x, y).
+    fn quad_to(&mut self, cx0: T, cy0: T, x: T, y: T);
+
+    /// Emit a cubic bezier segment from the current point with control
+    /// points at (cx0, cy0) and (cx1, cy1) and ending at (x, y).
+    fn curve_to(&mut self, cx0: T, cy0: T, cx1: T, cy1: T, x: T, y: T);
```

Comment:
My question here becomes: what exactly is the scope of this interface? do we anticipate supporting cubics? Are we converting multiple contours into a single path, by adding move/close?

Reply:
We'll need to support cubics when we add the CFF table. I know there is also a desire to add cubics to glyf, so I don't see any reason to leave this out.

It's worth mentioning that this is the trait that is used to extract glyphs in punchcut as part of our new callback-only API, so it needs to be general.

Reply:
And yes, it combines multiple contours with move/close. This is essentially the same as `FT_Outline_Decompose`.

Reply:
I suggest a comment re FT_O_D

Reply:
We should support cubic, needed for both CFF and hopefully in future glyf as well (https://github.com/harfbuzz/boring-expansion-spec, 2nd point)